nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/ui/common/navigation/navigation.component.ts

```diff
@@ -94,10 +114,10 @@ export class Navigation {
         );
     }
 
-    toggleTheme(value = !this.isDarkMode) {
-        this.isDarkMode = value;
-        this._document.body.classList.toggle('dark-theme', value);
-
-        // TODO: save to local storage or read OS settings???
+    toggleTheme(theme: string) {
+        this.theme = theme;
+        this.storage.setItem('theme', theme);
+        this.darkModeOn = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
```

Comment:
Does `darkModeOn` need to be re-assigned here? It is currently being assigned in the listener above so I suspect it should be always up to date.